src/api/graphapi.h

```diff
@@ -0,0 +1,417 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Apache License, Version 2.0,
+* modified with the Commons Clause restriction.
+*/
+
+#ifndef REDISGRAPH_API_H
+#define REDISGRAPH_API_H
+
+// #ifdef DREDIS_MODULE_TARGET
+
+// RG_VERSION: a single integer for comparing spec and version levels
+#define RG_VERSION(major, minor, sub) \
+    (((major)*1000ULL + (minor))*1000ULL + (sub))
+
+// The version of this implementation, and the RedisGraph API version:
+#define RG_DATE "Jan 28, 2019"
+#define RG_MAJOR 1
+#define RG_MINOR 0
+#define RG_SUB   0
+
+//------------------------------------------------------------------------------
+// Include files required by RedisGraph
+//------------------------------------------------------------------------------
+#include <stdint.h>
+#include <stdio.h>
+
+//------------------------------------------------------------------------------
+// The RedisGraph entities
+//------------------------------------------------------------------------------
+typedef uint64_t RG_Node ;
+typedef uint64_t RG_Edge ;
+
+//------------------------------------------------------------------------------
+// the RedisGraph integer
+//------------------------------------------------------------------------------
+typedef uint64_t RG_Index ;
+
+//------------------------------------------------------------------------------
+// RedisGraph error and informational codes
+//------------------------------------------------------------------------------
+
+// All RedisGraph functions return a code that indicates if it was successful
+// or not.
+
+typedef enum
+{
+    RG_SUCCESS,             // all is well
+    RG_NO_VALUE,            // value requested but not there
+}
+RG_Info ;
+
+/* High level API Functions */
+typedef struct RedisModuleCtx RedisModuleCtx;
+typedef struct RedisModuleString RedisModuleString;
+RG_Info RG_init                     // start up RedisGraphAPI
+(
+    RedisModuleCtx *ctx,            // redis module context
+    RedisModuleString *graphid      // graph id
+) ;
+
+RG_Info RG_finalize ( ) ;           // finish RedisGraphAPI
+
```

Comment:
I think schema-style interfaces will be necessary: enumerate labels / relation types, enumerate all properties on a given label/type.